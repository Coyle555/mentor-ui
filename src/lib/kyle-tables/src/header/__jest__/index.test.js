import React from 'react';
import { Header } from '../index';
import renderer from 'react-test-renderer';

test('Render of header where is it not collapsed', () => {
	global.getComputedStyle = () => ({ width: 1280 });
	global.innerWidth = 1280;

	const tree = renderer.create(<Header />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Render of header where it is collapsed', () => {
	global.getComputedStyle = () => ({ width: 600 });
	global.innerWidth = 600;

	const tree = renderer.create(<Header />).toJSON();

	expect(tree).toMatchSnapshot();
});
