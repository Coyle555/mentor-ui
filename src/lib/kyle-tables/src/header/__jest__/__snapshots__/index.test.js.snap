// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render of header where is it not collapsed 1`] = `
Array [
  <div
    style={
      Object {
        "display": "none",
        "height": 0,
        "position": "absolute",
        "visibility": "hidden",
        "width": 0,
      }
    }
  />,
  <div
    className="table-header-container"
  >
    <div
      className="table-header-filter"
    >
      <div
        className="filter-tokenizer"
      >
        <span
          className="input-group-addon left-addon"
          data-for="structured-query-tooltip"
          data-tip="Save Search"
          onClick={[Function]}
          style={
            Object {
              "background": "lightgrey",
              "cursor": "not-allowed",
            }
          }
        >
          <i
            className="fal fa-save"
          />
        </span>
        <span
          className="input-group-addon left-addon"
          data-for="structured-query-tooltip"
          data-tip="View Filters"
          onClick={[Function]}
          style={
            Object {
              "background": "lightgrey",
              "borderRadius": 0,
              "cursor": "not-allowed",
              "position": "relative",
            }
          }
        >
          <i
            className="far fa-list"
          />
          <span
            className="active-filter-count"
          >
            0
          </span>
        </span>
        <div
          className="token-collection"
          onClick={[Function]}
        >
          <div
            className="filter-input-group"
          >
            <div
              className="filter-category"
            >
              
            </div>
            <div
              className="filter-operator"
            >
              
            </div>
            <div
              className="typeahead"
            >
              <input
                className="filter-tokenizer-text-input"
                disabled={false}
                onChange={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
          id="structured-query-tooltip"
        />
      </div>
    </div>
    <div
      className="table-header-toolbar"
    >
      <span
        data-for="table-tooltip"
        data-tip="Add Record"
      >
        <button
          className="btn-table"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fal fa-file-plus"
          />
        </button>
      </span>
      <span
        data-for="table-tooltip"
        data-tip="Add Multiple Records"
      >
        <button
          className="btn-table"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fal fa-copy"
          />
        </button>
      </span>
      <span
        data-for="table-tooltip"
        data-tip="Toggle Columns"
      >
        <button
          className="btn-table"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fal fa-columns"
          />
        </button>
      </span>
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
        id="table-tooltip"
      />
    </div>
  </div>,
]
`;

exports[`Render of header where it is collapsed 1`] = `
Array [
  <div
    style={
      Object {
        "display": "none",
        "height": 0,
        "position": "absolute",
        "visibility": "hidden",
        "width": 0,
      }
    }
  />,
  <div
    className="table-header-container"
  >
    <div
      className="table-header-filter"
    >
      <div
        className="filter-tokenizer"
      >
        <span
          className="input-group-addon left-addon"
          data-for="structured-query-tooltip"
          data-tip="Save Search"
          onClick={[Function]}
          style={
            Object {
              "background": "lightgrey",
              "cursor": "not-allowed",
            }
          }
        >
          <i
            className="fal fa-save"
          />
        </span>
        <span
          className="input-group-addon left-addon"
          data-for="structured-query-tooltip"
          data-tip="View Filters"
          onClick={[Function]}
          style={
            Object {
              "background": "lightgrey",
              "borderRadius": 0,
              "cursor": "not-allowed",
              "position": "relative",
            }
          }
        >
          <i
            className="far fa-list"
          />
          <span
            className="active-filter-count"
          >
            0
          </span>
        </span>
        <div
          className="token-collection"
          onClick={[Function]}
        >
          <div
            className="filter-input-group"
          >
            <div
              className="filter-category"
            >
              
            </div>
            <div
              className="filter-operator"
            >
              
            </div>
            <div
              className="typeahead"
            >
              <input
                className="filter-tokenizer-text-input"
                disabled={false}
                onChange={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
          id="structured-query-tooltip"
        />
      </div>
    </div>
    <div
      className="table-header-toolbar"
    >
      <span
        data-for="table-tooltip"
        data-tip="Add Record"
      >
        <button
          className="btn-table"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fal fa-file-plus"
          />
        </button>
      </span>
      <span
        data-for="table-tooltip"
        data-tip="Add Multiple Records"
      >
        <button
          className="btn-table"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fal fa-copy"
          />
        </button>
      </span>
      <span
        data-for="table-tooltip"
        data-tip="Toggle Columns"
      >
        <button
          className="btn-table"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fal fa-columns"
          />
        </button>
      </span>
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
        id="table-tooltip"
      />
    </div>
  </div>,
]
`;
